sw/device/silicon_creator/lib/mock_crc32.h

```diff
@@ -0,0 +1,33 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_MOCK_CRC32_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_MOCK_CRC32_H_
+
+#include "sw/device/lib/base/global_mock.h"
+#include "sw/device/silicon_creator/lib/crc32.h"
+
+namespace mask_rom_test {
+namespace internal {
+
+/**
+ * Mock class for crc32.
+ */
+class MockCrc32 : public global_mock::GlobalMock<MockCrc32> {
+ public:
+  MOCK_METHOD(void, Init, (uint32_t *));
+  MOCK_METHOD(void, Add8, (uint32_t *, uint8_t));
+  MOCK_METHOD(void, Add32, (uint32_t *, uint32_t));
+  MOCK_METHOD(void, Add, (uint32_t *, const void *, size_t));
+  MOCK_METHOD(uint32_t, Finish, (const uint32_t *));
+  MOCK_METHOD(uint32_t, Crc32, (const void *, size_t));
+};
+
+}  // namespace internal
+
+using MockCrc32 = testing::StrictMock<internal::MockCrc32>;
```

Comment:
nit: No change needed, but just wanted to mention that the style guide has something to say regarding exporting names in header files. More details here: https://abseil.io/tips/119

Reply:
Thanks for the link! I'll definitely read through that page and references to the style guide. Yeah, this is not ideal but we need to define our mocks in a [header](https://google.github.io/googletest/gmock_for_dummies.html#where-to-put-it) so that multiple tests can share the same definition and most of the time we want them to be `StrictMock`s (with a few exceptions where we use `NiceMock`s) so this alias is indeed intended to be used by the clients, i.e. our unit tests.